cns/NetworkContainerContract.go

```diff
@@ -131,9 +132,22 @@ func (req *CreateNetworkContainerRequest) Validate() error {
 	if _, err := uuid.Parse(strings.TrimPrefix(req.NetworkContainerid, SwiftPrefix)); err != nil {
 		return errors.Wrapf(ErrInvalidNCID, "NetworkContainerID %s is not a valid UUID: %s", req.NetworkContainerid, err.Error())
 	}
+	if req.PrimaryInterfaceIdentifier != "" && !isValidIP(req.PrimaryInterfaceIdentifier) {
+		return errors.Wrapf(ErrInvalidIP, "PrimaryInterfaceIdentifier %s is not a valid ip address", req.PrimaryInterfaceIdentifier)
+	}
+	if req.IPConfiguration.GatewayIPAddress != "" && !isValidIP(req.IPConfiguration.GatewayIPAddress) {
+		return errors.Wrapf(ErrInvalidIP, "GatewayIPAddress %s is not a valid ip address", req.IPConfiguration.GatewayIPAddress)
+	}
 	return nil
 }
 
+func isValidIP(ipStr string) bool {
+	if ip, _, err := net.ParseCIDR(ipStr); err == nil {
+		return ip != nil
+	}
+	return net.ParseIP(ipStr) != nil
```

Comment:
This block is a little hard to follow (because it's trying to be so compact). I think readability should be preferred here, where I need to be able to quickly grok what the valid/invalid inputs are and don't necessarily remember what `net.ParseIP` returns - is it an `error`? Is it a `net.IP` or a `*net.IP`? The answer significantly changes my interpretation of this function and its correctness.